### DISCUSSION TITLE
QNN backend

### DISCUSSION THREAD

**Discussion by Francis235** [ggml-org/llama.cpp#9693] [Q&A]
Hello, everyone. How is llama.cpp's support for qualcomm QNN backend now? Is there any usage tutorials?

**Comment by ExtReMLapin** [comment]
https://github.com/ggerganov/llama.cpp/pull/9662

Closed PR : https://github.com/ggerganov/llama.cpp/pull/6869

**Reply by Francis235** [reply]
https://github.com/ggerganov/llama.cpp/pull/9662 is still not merged, have you done any verification? How should I use QNN-backend in llama.cpp?

**Comment by FranzKafkaYu** [comment]
there is a little hope for QNN backend in llama.cpp,I found that there is a project which you may find useful:https://github.com/UbiquitousLearning/mllm

**Reply by bingo787** [reply]
mlllm run would codedump on the Qualcomm Soc

**Comment by bingo787** [comment]
> Hello, everyone. How is llama.cpp's support for qualcomm QNN backend now? Is there any usage tutorials?

The call of a single operator is very inefficient and must be optimized from the model to build a static graph on QNN-backend
Maybe you can use qualcomm Genie SDK.

**Reply by Francis235** [reply]
Thanks for you kindly reply, I am trying both mllm and Genie SDK now.
